docs/new-style-script-objects.md

```diff
@@ -36,3 +38,19 @@ The equivalent functionality can be accomplished asynchronously:
                 });
         }
     };
+
+As an example of the utility of pre/post functionality in a new-style script
+object bundle, here is an example of how to use `ChromeUtils.collectPerfStats`:
+
+    module.exports = {
+        requires: { privilege: true },
+        pre: function() {
+                ChromeUtils.setPerfStatsCollectionMask(255);
+        },
+        collect: async function() {
+                const stats = await ChromeUtils.collectPerfStats();
```

Comment:
I'm sorry but I'm confused by what you are saying here. I *think* that the special semantics of browsertime function execution negate what you are saying. Once you clarify, I will attempt to figure something out :-)

Reply:
You had written something like:
```
async function() {
  const stats = await foo();
  return Promise(r => r(f(stats)));
}
```
There's no need to handle the `Promise` wrapping yourself, that's what `async function` does.  Just `return f(stats)`; that will be a `Promise`.

Reply:
It's actually odder than this because of the way that these functions are executed throughout the browsertime codebase. No matter what you return (Promise or value), it will be a value as the result. Promises (as return values) are all collapsed (by following all the thenables). 

I made it specifically a Promise() because of the way that the `async` keyword is stripped when I serialize the function for transmission to the webdriver. I investigated this last night and (see above) the outcome is ... it doesn't matter :-)